Components: change shift behavior of CueButton and PlayButton

Be-ing opened the pull request:
shift + PlayButton: play in reverse while held
shift + CueButton: jump to beginning of track and stop

Previously shift + CueButton was unused. Looking at the documentation for other mappings, I see that many people want a reverse button. I think this is an intuitive way to map these behaviors.

Be-ing committed:
Components: change shift behavior of CueButton and PlayButton

shift + PlayButton: play in reverse while held
shfit + CueButton: jump to beginning of track and stop

Be-ing committed:
Components small code cleanup

uklotzde commented on res/controllers/midi-components-0.0.js:
```diff
@@ -181,9 +181,11 @@
         shift: function () {
-            this.inKey = 'start_stop';
+            this.inKey = 'reverse';
+            this.onlyOnPress = false;
```
inValueScale() still returns the opposite/inverse of the current value, i.e. it toggles the value. As far as I can imagine this should have the same effect as setting it to 1 when pressed and to 0 when released. That's what I called a 'switch' button. The 'toggle' behaviour used to simulate a 'switch' here will get out of sync if 'reverse' ever contains an unexpected value or if one button event is lost somehow. Setting the value directly depending on the button event would be easier to understand and also more reliable.

uklotzde replied:
The deck might already be playing in reverse when you press shift + play. In this case the deck should continue playing in reverse and continue playing forward when releasing the button. Your implementation toggles reverse both when pressing and releasing the button, which has the opposite effect.

Be-ing replied:
Thanks for point out this issue.

Be-ing committed:
Components: fixes for PlayButton and CueButton

Be-ing commented:
Currently, pressing shift + play if the deck is stopped does not start playing the deck in reverse. The script would have to toggle the `play` CO to do that, but that would not be guaranteed to work correctly because toggling two COs simultaneously via script creates a race condition. I think the behavior of the `reverse` CO should be modified so it starts playback when set to 1 if `play` is 0. If `reverse` is set to `0` before `play` is set to 1 by something else (like if reverse and play were mapped to separate buttons), that would stop playback.

Be-ing committed:
Components: let PlayButton and CueButton work with unusual inPress functions

Using inValueScale presumed that the third byte of the MIDI message could be used to distinguish a button press from a button release, but that is not necessarily the case for every controller.

uklotzde commented:
Thank you for the updates, @Be-ing! I remember that I had to use similar workarounds for switching between unshift/shift behaviour, because you never know which of the two buttons is released first. LGTM, even if I'm not able to test it now.

daschuer commented:
Thank you.
